### PR TITLE
Chore/Upgrade llamaindex

### PR DESCRIPTION
**Pull request by HenryHengZJ**
- llamaindex version upgrade
- openai agent
- querytool fix

### CODE REVIEW

File: packages/components/nodes/chatmodels/ChatAnthropic/ChatAnthropic_LlamaIndex.ts [hunk 1]
```diff
@@ -39,22 +39,22 @@ class ChatAnthropic_LlamaIndex_ChatModels implements INode {
...
                     {
                         label: 'claude-3-opus',
-                        name: 'claude-3-opus-20240229',
+                        name: 'claude-3-opus',
                         description: 'Most powerful model for highly complex tasks'
                     },
                     {
                         label: 'claude-3-sonnet',
-                        name: 'claude-3-sonnet-20240229',
+                        name: 'claude-3-sonnet',
```
Comment (vinodkiran) [inline packages/components/nodes/chatmodels/ChatAnthropic/ChatAnthropic_LlamaIndex.ts:47]: The model names are different in the LLamaIndex and Langchain Nodes.

File: packages/components/nodes/chatmodels/ChatAnthropic/ChatAnthropic_LlamaIndex.ts [hunk 2]
```diff
@@ -39,22 +39,22 @@ class ChatAnthropic_LlamaIndex_ChatModels implements INode {
...
                         description: 'Claude 2 latest major version, automatically get updates to the model as they are released'
                     },
                     {
-                        label: 'claude-instant-1',
-                        name: 'claude-instant-1',
+                        label: 'claude-instant-1.2',
+                        name: 'claude-instant-1.2',
                         description: 'Claude Instant latest major version, automatically get updates to the model as they are released'
                     }
                 ],
```
Comment (vinodkiran) [inline packages/components/nodes/chatmodels/ChatAnthropic/ChatAnthropic_LlamaIndex.ts:60]: How about Claude Haiku?

File: packages/components/nodes/chatmodels/ChatAnthropic/ChatAnthropic_LlamaIndex.ts (hunk 1 above)
Comment (HenryHengZJ) [reply packages/components/nodes/chatmodels/ChatAnthropic/ChatAnthropic_LlamaIndex.ts:47]: yes its a constants [here](https://github.com/run-llama/LlamaIndexTS/blob/main/packages/core/src/llm/LLM.ts#L631)

File: packages/components/nodes/chatmodels/ChatAnthropic/ChatAnthropic_LlamaIndex.ts (hunk 2 above)
Comment (HenryHengZJ) [reply packages/components/nodes/chatmodels/ChatAnthropic/ChatAnthropic_LlamaIndex.ts:60]: https://github.com/run-llama/LlamaIndexTS/blob/main/packages/core/src/llm/LLM.ts#L631

**Pull request merged**